### PR TITLE
LibJS: Add %TypedArray%.prototype.toReversed 

### PR DESCRIPTION
**Pull request by beesaferoot**

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp [hunk 1]
```diff
@@ -1512,6 +1513,39 @@ JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::to_locale_string)
     return js_string(vm, builder.to_string());
 }
 
+// 1.2.2.1.3 %TypedArray%.prototype.toReversed
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp:1516]: Missing spec URL

File: Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp [hunk 2]
```diff
@@ -1512,6 +1513,39 @@ JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::to_locale_string)
...
+
+          // d. Perform ! Set(A, Pk, kValue, true).
+          MUST(return_array->set(k, from_value, Object::ShouldThrowExceptions::Yes));
+          // e. Set k to k + 1.
+    }
+
+    return return_array;
+}
+
 // 1.2.2.1.4 %TypedArray%.prototype.toSorted ( comparefn ) https://tc39.es/proposal-change-array-by-copy/#sec-%typedarray%.prototype.toSorted
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp:1549]: Please also add `, ` between the signature and the URL here while you're at it :)

File: Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp [hunk 3]
```diff
@@ -1512,6 +1513,39 @@ JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::to_locale_string)
...
+
+          // c. Let fromValue be ! Get(O, from).
+          auto from_value = MUST(typed_array->get(from));
+
+          // d. Perform ! Set(A, Pk, kValue, true).
+          MUST(return_array->set(k, from_value, Object::ShouldThrowExceptions::Yes));
+          // e. Set k to k + 1.
+    }
+
+    return return_array;
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp:1546]: `// 7. Return A.`

File: Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp [hunk 4]
```diff
@@ -1512,6 +1513,39 @@ JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::to_locale_string)
...
+    // 6. Repeat, while k < length
+    for(size_t k = 0; k < length; k++){
+          // a. Let from be ! ToString(𝔽(length - k - 1)).
+          // Let Pk be ! ToString(𝔽(k)).
+          auto from = length - k - 1;
+
+          // c. Let fromValue be ! Get(O, from).
+          auto from_value = MUST(typed_array->get(from));
+
+          // d. Perform ! Set(A, Pk, kValue, true).
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp:1541]: aha, a spec issue! (`kValue` -> `fromValue`). I'll file a fix in their repo :+1:

File: Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp [hunk 5]
```diff
@@ -58,6 +58,7 @@ void TypedArrayPrototype::initialize(GlobalObject& object)
     define_native_function(vm.names.map, map, 1, attr);
     define_native_function(vm.names.toLocaleString, to_locale_string, 0, attr);
     define_native_function(vm.names.toSorted, to_sorted, 1, attr);
+    define_native_function(vm.names.toReversed, to_reversed, 0, attr);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp:61]: Let's put this before toSorted

File: Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.h [hunk 6]
```diff
@@ -53,6 +53,7 @@ class TypedArrayPrototype final : public Object {
     JS_DECLARE_NATIVE_FUNCTION(map);
     JS_DECLARE_NATIVE_FUNCTION(to_locale_string);
     JS_DECLARE_NATIVE_FUNCTION(to_sorted);
+    JS_DECLARE_NATIVE_FUNCTION(to_reversed);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.h:56]: Let's put this before toSorted

**Review by linusg — COMMENTED** [review]
Nice, mostly LGTM! First commit contains the tests that should probably go into the second :)

File: Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp [hunk 7]
```diff
@@ -1512,7 +1513,41 @@ JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::to_locale_string)
     return js_string(vm, builder.to_string());
 }
 
-// 1.2.2.1.4 %TypedArray%.prototype.toSorted ( comparefn ) https://tc39.es/proposal-change-array-by-copy/#sec-%typedarray%.prototype.toSorted
+// 1.2.2.1.3 %TypedArray%.prototype.toReversed, https://tc39.es/proposal-change-array-by-copy/#sec-%typedarray%.prototype.toReversed
```
[the remark this answers is not in the document]
Comment (beesaferoot) [reply Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp:1516]: Oops, sorry I missed that. I will update, thanks.

File: Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp [hunk 8]
```diff
@@ -1512,7 +1513,41 @@ JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::to_locale_string)
...
+    MarkedVector<Value> arguments(vm.heap());
+    arguments.empend(length);
+    // 4. Let A be ? TypedArrayCreateSameType(O, « 𝔽(length) »).
+    auto* return_array = TRY(typed_array_create_same_type(global_object, *typed_array, move(arguments)));
+
+    // 5. Let k be 0.
+    // 6. Repeat, while k < length
+    for (size_t k = 0; k < length; k++) {
+        // a. Let from be ! ToString(𝔽(length - k - 1)).
+        // Let Pk be ! ToString(𝔽(k)).
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp:1535]: Missing b.

Comment (beesaferoot) [reply Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp]: Added. :)

**Pull request merged**